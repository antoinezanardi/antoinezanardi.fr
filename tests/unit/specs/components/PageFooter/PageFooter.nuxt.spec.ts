import type { mount } from "@vue/test-utils";
import type { ComponentMountingOptions } from "@vue/test-utils/dist/mount";

import { PageFooter } from "#components";
import { mountSuspendedComponent } from "@tests/unit/utils/helpers/mount.helpers";

describe("Page Footer Component", () => {
  let wrapper: ReturnType<typeof mount<typeof PageFooter>>;

  async function mountPageFooterComponent(options: ComponentMountingOptions<typeof PageFooter> = {}): Promise<ReturnType<typeof mount<typeof PageFooter>>> {
    return mountSuspendedComponent(PageFooter, { ...options });
  }

  beforeEach(async() => {
    wrapper = await mountPageFooterComponent();
  });

  it("should match snapshot when rendered.", () => {
    expect(wrapper).toBeTruthy();
    expect(wrapper.html()).toMatchSnapshot();
  });

  describe("Text", () => {
    it("should display Antoine ZANARDI and full year when rendered.", () => {
      const title = wrapper.find<HTMLDivElement>(".title");

      expect(title.text()).toBe("Antoine ZANARDI - 2022");
    });
  });
});