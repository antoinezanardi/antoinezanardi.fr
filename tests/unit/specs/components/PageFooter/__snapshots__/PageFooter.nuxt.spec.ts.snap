// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Page Footer Component > should match snapshot when rendered. 1`] = `
"<footer class="footer py-1">
  <div class="container text-center" data-aos="fade-up" data-aos-anchor-placement="top-bottom">
    <linked-in-button-stub class="me-2"></linked-in-button-stub>
    <git-hub-button-stub></git-hub-button-stub>
  </div>
  <div class="font-italic h4 my-2 text-center text-muted title"><span>Antoine Zanardi</span><span> - 2022</span></div>
</footer>"
`;
